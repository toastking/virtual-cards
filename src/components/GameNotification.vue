<template>
  <div class="container">
    <section v-if="isYourTurn" class="hero is-info">
      <div class="hero-body">
        <div class="container">
          <h1 class="title">Your Turn!</h1>
        </div>
      </div>
    </section>
    <section v-if="gameOver" class="hero is-success">
      <div class="hero-body">
        <div class="container">
          <h1 class="title">Game Over!</h1>
        </div>
      </div>
    </section>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapGetters } from 'vuex';
export default Vue.extend({
  computed: { ...mapGetters(['isYourTurn', 'gameOver']) },
});
</script>