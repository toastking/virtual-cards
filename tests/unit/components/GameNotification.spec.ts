import GameNotification from '@/components/GameNotification.vue';
import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import { Store } from 'vuex-mock-store';

describe('GameNotification', () => {
  const store = new Store({ getters: { isYourTurn: false, gameOver: false } });
  // add other mocks here so they are accessible in every component
  const mocks = {
    $store: store,
  };

  const localVue = createLocalVue();
  localVue.use(Buefy);

  afterEach(() => {
    store.reset();
  });

  describe('snapshots', () => {
    test('shows nothing when its not the players turn', () => {
      store.getters.isYourTurn = false;
      store.getters.gameOver = false;
      const wrapper = mount(GameNotification, { mocks, localVue });
      expect(wrapper.element).toMatchSnapshot();
    });

    test('show player banner when it is their turn', () => {
      store.getters.isYourTurn = true;
      const wrapper = mount(GameNotification, { mocks, localVue });
      expect(wrapper.element).toMatchSnapshot();
    });

    test('shows game over banner when the game is over', () => {
      store.getters.isYourTurn = false;
      store.getters.gameOver = true;
      const wrapper = mount(GameNotification, { mocks, localVue });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  test('dispatches restart game action', () => {
    store.getters.gameOver = true;
    const wrapper = mount(GameNotification, { mocks, localVue });
    wrapper.find('#new-game-button').trigger('click');
    expect(store.dispatch).toHaveBeenCalledWith('restartGame');
  });
});
