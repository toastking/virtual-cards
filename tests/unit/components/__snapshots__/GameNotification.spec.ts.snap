// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameNotification snapshots show player banner when it is their turn 1`] = `
<section
  class="hero is-info container"
  enter-active-class="animate__animated animate__jackInTheBox"
  exit-active-class="animate__animated animate__slideOutLeft"
  tag="div"
>
  <div
    class="hero-body"
  >
    <div
      class="container"
    >
      <h1
        class="title"
      >
        Your Turn!
      </h1>
    </div>
  </div>
</section>
`;

exports[`GameNotification snapshots shows game over banner when the game is over 1`] = `
<section
  class="hero is-success container"
  enter-active-class="animate__animated animate__jackInTheBox"
  exit-active-class="animate__animated animate__slideOutLeft"
  tag="div"
>
  <div
    class="hero-body"
  >
    <div
      class="container"
    >
      <h1
        class="title"
      >
        Game Over!
      </h1>
       
      <button
        class="button is-success is-inverted"
        id="new-game-button"
        type="button"
      >
        <span
          class="icon is-small"
        >
          <i
            class="mdi mdi-redo"
          />
        </span>
         
        <span>
          New Game
        </span>
         
        <!---->
      </button>
    </div>
  </div>
</section>
`;

exports[`GameNotification snapshots shows nothing when its not the players turn 1`] = `<!---->`;
